.github/workflows/windows-amd64.yml

```diff
@@ -0,0 +1,655 @@
+name: "Windows amd64"
+on:
+  pull_request:
+  push:
+    branches:
+      - master
+env:
+  TASKCLUSTER_TASK_DIR: ${{ github.workspace }}
+  TASKCLUSTER_ARTIFACTS: ${{ github.workspace }}/artifacts
+  TC_MSYS_VERSION: MSYS_NT-10.0-17763
+defaults:
+  run:
+    shell: bash
+jobs:
+  swig_Windows:
+    name: "Build SWIG for Windows"
+    runs-on: windows-2019
+    steps:
+      - name: Switch git-bash shell to MSYS2 shell by adding MSYS2 path to PATH front
+        run: echo "D:\a\_temp\msys\msys64\usr\bin" >> $GITHUB_PATH
+      - uses: mozilla/setup-msys2@v2
+        with:
+          msystem: MSYS
+          update: true
+          install: >-
+            autoconf
+            automake
+            bison
+            gcc
+            git
+            make
+            pcre-devel
+      - uses: actions/checkout@v2
+        with:
+          repository: "swig/swig"
+          ref: "90cdbee6a69d13b39d734083b9f91069533b0d7b"
+      - run: |
+          mkdir -p build-static/
+      - run: |
+          sh autogen.sh
+          ./configure \
+            --prefix=`pwd`/build-static/ \
+            --program-prefix=ds-
+      - run: |
+          make -j
+      - run: |
+          make install
+      - uses: actions/upload-artifact@v2
+        with:
+          name: ${{ github.job }}
+          path: ${{ github.workspace }}/build-static/
+  swig_Linux:
+    name: "Build SWIG for Linux"
+    runs-on: ubuntu-20.04
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          repository: "swig/swig"
+          ref: "90cdbee6a69d13b39d734083b9f91069533b0d7b"
+      - run: |
+          sudo apt-get install -y --no-install-recommends autoconf automake build-essential bison libpcre2-dev
+      - run: |
+          mkdir -p build-static/
+      - run: |
+          sh autogen.sh
+          ./configure \
+            --prefix=${{ github.workspace }}/build-static/ \
+            --program-prefix=ds-
+      - run: |
+          make -j
+      - run: |
+          make install
+      - uses: actions/upload-artifact@v2
+        with:
+          name: ${{ github.job }}
+          path: ${{ github.workspace }}/build-static/
+  build-ctc-decoder-windows:
+    name: "Build CTC decoder Windows Python package for testing"
+    needs: [swig_Windows]
+    runs-on: windows-2019
+    steps:
+      - name: Switch git-bash shell to MSYS2 shell by adding MSYS2 path to PATH front
+        run: echo "D:\a\_temp\msys\msys64\usr\bin" >> $GITHUB_PATH
+      - uses: mozilla/setup-msys2@v2
+        with:
+          msystem: MSYS
+          path-type: inherit
+          update: true
+          install: >-
+            git
+            make
```

Comment:
can't we rely on `nmake` from msvc? it might require a bit of adaptations in `native_client/definitions.mk`, but it would make it cleaner / easier for contributors

Reply:
Not sure. In general the Windows setup leads to tons of PATH conflicts and I'm not sure maintaining `make` and `nmake` compatibility is actually easier for contributors.